src/apps/handlers/prepareMovementData.js

```diff
@@ -1,10 +1,21 @@
 export default function getMovementData(data) {
   // eslint-disable-next-line no-unused-vars
-  const speeds = Object.entries(data.system.attributes.movement).filter(([_, speed]) => speed);
+  const hover = data.system.attributes.movement.traits.hover
+  const speeds = Object.entries(data.system.attributes.movement).filter(([key, speed]) => {
+    if (key === "fly" && !speed && hover) { return true; }
+    if (key === "traits") { return false; }
+    return speed
+  })
+
   const unit = game.i18n.localize('A5E.MeasurementFeetAbbr');
+  const hoverAbbr = game.i18n.localize('A5E.MovementHoverAbbr');
 
   return speeds.map(([name, speed]) => {
     const label = game.i18n.localize(CONFIG.A5E.movement[name]);
-    return `${label} - ${speed} ${unit}`;
+    if (name == "fly" && hover) {
```

Comment:
I'd like a line break here to separate the conditional block from the variable definitions. Makes it much easier to spot at a glance that these are unrelated pieces of code.